### ISSUE TITLE
Encode transaction JSON object

### ISSUE THREAD

**Issue by olemis** [skycoin/skycoin#2395]
**Feature description**

Get raw transaction from JSON

**Is your feature request related to a problem? Please describe.**
While running pre-release testing procedure of hardware wallet libraries it is necessary to test signing with the device. After doing so and adding signatures in JSON object representing the target transaction , there is no way to encode it to raw so as to broadcast it.

**Describe the solution you'd like**
Facilitate transaction serialization to users 

**Describe alternatives you've considered**
[`/api/v2/transaction/sign`](https://github.com/skycoin/skycoin/tree/master/src/api/README.md#sign-transaction) is not appropriate since signing is performed by the node.

**Additional context**
Related to skycoin/hardware-wallet-js#112 .

**Possible implementation**

- [x] CLI command to encode transaction JSON object
- [x] (Optionally) update transaction inner and outer hash (see skycoin/hardware-wallet-go#144)
- ~REST API endpoint to encode transaction JSON object~

**Comment by olemis** [comment]
While working on skycoin/hardware-wallet-go#144 had to apply this (incorrect) patch against e2db0000541af05632174970ed7b7bf1afcbf4b8 so as to be able to finish the process .

```diff

diff --git a/src/api/spend.go b/src/api/spend.go
index 9c91ae556..e0e2e1950 100644
--- a/src/api/spend.go
+++ b/src/api/spend.go
@@ -190,8 +190,10 @@ func (r *CreatedTransaction) ToTransaction() (*coin.Transaction, error) {
                return nil, err
        }

-       if t.Hash() != hash {
-               return nil, errors.New("readable.Transaction.Hash does not match parsed transaction hash")
+       if false {
+               if t.Hash() != hash {
+                       return nil, errors.New("readable.Transaction.Hash does not match parsed transaction hash")
+               }
        }

        return &t, nil
```

Need a way to relax such constraints by updating txn hashes .

**Comment by jdknives** [comment]
@therealssj has this been fixed by #2468 ?

**Comment by olemis** [comment]
> @therealssj has this been fixed by #2468 ?

yes
